### PR TITLE
Differential evolution hyperparameter changes

### LINKED ISSUE #362: Differential Evolution advice on hyper-parameters
The Differential Evolution MCMC has two hyper-parameters gamma and the normal scale coefficient. There should be some advice in the documentation (http://pints.readthedocs.io/en/latest/mcmc_samplers/differential_evolution_mcmc.html) on how to choose these, or what their minimum/maximum bounds are.

@ben18785

### PR DESCRIPTION
**Pull request by ben18785**
Closes #362 

I have added new methods for DE including allowing a uniform (instead of a normal) proposal distribution; intermittent setting of gamma=1 every set number of iterations and an option to use absolute rather than relative scaling for b.

Importantly, I also changed the default value of b=0.001. It seems that this makes the algorithm (at least in the notebook example) run more effectively. Clearly small values of this parameter are desirable.

### CODE REVIEW

File: pints/_mcmc/_differential_evolution.py [hunk 1]
```diff
@@ -60,33 +67,65 @@ def __init__(self, chains, x0, sigma0=None):
...
         return self._proposed
 
+    def set_normal_error(self, normal_error):
+        """
+        If true sets the error process to be a normal
+        error, N(0, b*); if false, it uses a uniform error
+        U(-b*, b*); where b* = b if absolute scaling used
+        and b* = mu * b if relative used instead
+        """
+        normal_error = bool(normal_error)
```
Comment (MichaelClerx) [inline pints/_mcmc/_differential_evolution.py]: Can be 1 line!

File: pints/_mcmc/_differential_evolution.py [hunk 2]
```diff
@@ -60,33 +67,65 @@ def __init__(self, chains, x0, sigma0=None):
...
+
             # Set as read only
             self._proposed.setflags(write=False)
 
         # Return proposed points
         return self._proposed
 
+    def set_normal_error(self, normal_error):
+        """
+        If true sets the error process to be a normal
```
Comment (MichaelClerx) [inline pints/_mcmc/_differential_evolution.py]: Please break lines at 79 chars! (You can probably set up your editor to display a gray line or something)

File: pints/_mcmc/_differential_evolution.py [hunk 3]
```diff
@@ -158,14 +229,14 @@ def tell(self, proposed_log_pdfs):
         self._current.setflags(write=False)
         return self._current
 
-    def set_normal_scale_coefficient(self, b):
+    def set_scale_coefficient(self, b):
         """
-        Sets the normal scale coefficient ``b`` used in updating the position
-        of each chain.
+        Sets the scale coefficient ``b`` of error process used in updating
```
Comment (MichaelClerx) [inline pints/_mcmc/_differential_evolution.py]: the error process

File: pints/_mcmc/_differential_evolution.py [hunk 4]
```diff
@@ -189,7 +260,9 @@ def set_hyper_parameters(self, x):
         See :meth:`TunableMethod.set_hyper_parameters()`.
         """
         self.set_gamma(x[0])
-        self.set_normal_scale_coefficient(x[1])
+        self.set_scale_coefficient(x[1])
```
Comment (MichaelClerx) [inline pints/_mcmc/_differential_evolution.py]: Please update the list of hyperparameters in the docstring above.

**Comment by MichaelClerx** [comment]
Looks good! (The failing test is an issue with a new version of Sphinx, which we've fixed on master, so merging will resolve this).
Only thing missing is full test cover (so we'd need to run a test with different scaling and one with normal errors, I think): https://codecov.io/gh/pints-team/pints/pull/497/diff?src=pr&el=tree#diff-cGludHMvX21jbWMvX2RpZmZlcmVudGlhbF9ldm9sdXRpb24ucHk=

**Comment by martinjrobins** [comment]
@ben18785: I've added `relative_scaling` to the list of hyper-parameters, assuming it is one. Let me know if this assumption was wrong!

**Pull request merged**